Framework/DataHandling/inc/MantidDataHandling/LoadEventNexus.h

```diff
@@ -470,15 +484,20 @@ void adjustTimeOfFlightISISLegacy(::NeXus::File &file, T localWorkspace,
  *  @return true if successful
  */
 template <typename T>
-bool LoadEventNexus::runLoadInstrument(const std::string &nexusfilename,
-                                       T localWorkspace,
-                                       const std::string &top_entry_name,
-                                       Algorithm *alg) {
+bool LoadEventNexus::runLoadInstrument(
+    const std::string &nexusfilename, T localWorkspace,
+    const std::string &top_entry_name, Algorithm *alg,
+    const std::shared_ptr<Kernel::NexusHDF5Descriptor> &descriptor) {
```

Comment:
Can some of these functions that just **use** `descriptor` take a non-owning raw pointer? I think that would better describe ownership and make the default a clearer value (`nullptr`)

Reply:
It can go either way, passing `shared_ptr &` reflecting non-ownership and using the `bool` operator for non-initialized checks, or raw pointer + check for `nullptr`. The former builds on top of the latter. Let me know if I should go ahead and move to raw pointers, I'd be happy to accommodate.

Reply:
In general Mantid has way too many `std::shared_ptr`s floating around and will take any opportunity to use an object, reference, or non-owning pointer.

Are there cases where we can use the getter within the member function, or are all of these member functions `static`?

Reply:
Sounds good, raw pointer it is. `LoadEventNexus::loadEvents` uses the getter, while `runLoadInstrument` is `static` . I'm passing a reference in `loadEntryMetadata` as even though is `static` it's only used within `NexusHDF5Descriptor` derived classes, so it's never `nullptr`.